source/extensions/filters/udp/udp_proxy/session_filters/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -106,18 +106,23 @@ ReadFilterStatus ProxyFilter::onData(Network::UdpRecvData& data) {
   return ReadFilterStatus::StopIteration;
 }
 
-void ProxyFilter::onLoadDnsCacheComplete(const Common::DynamicForwardProxy::DnsHostInfoSharedPtr&) {
+void ProxyFilter::onLoadDnsCacheComplete(
+    const Common::DynamicForwardProxy::DnsHostInfoSharedPtr& host_info) {
   ENVOY_LOG(debug, "load DNS cache complete, continuing");
   ASSERT(circuit_breaker_ != nullptr);
   circuit_breaker_.reset();
 
   load_dns_cache_completed_ = true;
   read_callbacks_->continueFilterChain();
 
-  while (!datagrams_buffer_.empty()) {
-    BufferedDatagramPtr buffered_datagram = std::move(datagrams_buffer_.front());
-    datagrams_buffer_.pop();
-    read_callbacks_->injectDatagramToFilterChain(*buffered_datagram);
+  if (host_info && host_info->address()) {
+    while (!datagrams_buffer_.empty()) {
+      BufferedDatagramPtr buffered_datagram = std::move(datagrams_buffer_.front());
+      datagrams_buffer_.pop();
+      read_callbacks_->injectDatagramToFilterChain(*buffered_datagram);
+    }
+  } else {
+    ENVOY_LOG(debug, "DNS lookup failed");
```

Comment:
I'm curious, if we simply LOG here (and don't explicitly set some sort of error condition) do we run the risk of getting ourselves in a state where we're just kinda stalled? Or does it all work out cleanly?

Reply:
@RyanTheOptimist Changed the approach for the fix, can you take a look, please?